modules/consul-cluster/main.tf

```diff
@@ -170,6 +172,8 @@ resource "aws_iam_instance_profile" "instance_profile" {
 }
 
 resource "aws_iam_role" "instance_role" {
+  count = "${var.enable_iam_setup}"
+
   name_prefix        = "${var.cluster_name}"
   assume_role_policy = "${data.aws_iam_policy_document.instance_role.json}"
```

Comment:
This should now be `element(concat(data.aws_iam_policy_document.instance_role.*.json, list("")), 0)`

Reply:
Ah yes, this would have come up if i could test the IAM enabled mode.  I will correct this.

Reply:
I have just removed the count from the data resource as that was un-required due to it not needing any IAM rights to do so.  That makes this change un-required also.